xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -419,70 +566,114 @@ private LdsUpdate processServerSideListener(Listener listener)
 
   @VisibleForTesting
   static StructOrError<Map<String, FilterConfig>> parseOverrideFilterConfigs(
-      Map<String, Any> rawFilterConfigMap) {
+      Map<String, Any> rawFilterConfigMap, FilterRegistry filterRegistry) {
     Map<String, FilterConfig> overrideConfigs = new HashMap<>();
     for (String name : rawFilterConfigMap.keySet()) {
       Any anyConfig = rawFilterConfigMap.get(name);
-      StructOrError<FilterConfig> filterConfig = parseRawFilterConfig(name, anyConfig, null, true);
-      if (filterConfig == null) {
-        continue;
+      String typeUrl = anyConfig.getTypeUrl();
+      boolean isOptional = false;
+      if (typeUrl.equals(TYPE_URL_FILTER_CONFIG)) {
+        io.envoyproxy.envoy.config.route.v3.FilterConfig filterConfig;
+        try {
+          filterConfig =
+              anyConfig.unpack(io.envoyproxy.envoy.config.route.v3.FilterConfig.class);
+        } catch (InvalidProtocolBufferException e) {
+          return StructOrError.fromError(
+              "FilterConfig [" + name + "] contains invalid proto: " + e);
+        }
+        isOptional = filterConfig.getIsOptional();
+        anyConfig = filterConfig.getConfig();
+        typeUrl = anyConfig.getTypeUrl();
       }
+      Message rawConfig = anyConfig;
+      if (anyConfig.getTypeUrl().equals(TYPE_URL_TYPED_STRUCT)) {
```

Comment:
`anyConfig.getTypeUrl()` should be same as `typeUrl` so no need to use the longer expression?

Reply:
Sounds good. Updated.